### ISSUE TITLE
t/local.t fails

### ISSUE THREAD

**Issue by eserte** [graphql-perl/GraphQL-Plugin-Convert-OpenAPI#2]
t/local.t started to fail:
```
Invalid JSON specification HASH(0x4810e3f0):
- /paths/~1enumtest/get/parameters/0: oneOf failed: oneOf failed: Properties not allowed: items, type. oneOf failed: (Not in enum list: header.. Properties not allowed: $ref.) (Not in enum list: formData.. Properties not allowed: $ref.) Properties not allowed: $ref. (Missing property.. Not in enum list: path.. Properties not allowed: $ref.) Properties not allowed: in, items, name, type. at /home/cpansand/.cpan/build/2018060612/JSON-Validator-2.08-1/blib/lib/JSON/Validator.pm line 179.
	JSON::Validator::load_and_validate_schema(JSON::Validator::OpenAPI::Mojolicious=HASH(0x47e8b630), "data://main/api.yaml", HASH(0x47e8b6f0)) called at /home/cpansand/.cpan/build/2018060612/JSON-Validator-2.08-1/blib/lib/JSON/Validator/OpenAPI.pm line 33
	JSON::Validator::OpenAPI::load_and_validate_schema(JSON::Validator::OpenAPI::Mojolicious=HASH(0x47e8b630), "data://main/api.yaml", HASH(0x47e8b6f0)) called at /home/cpansand/.cpan/build/2018060612/Mojolicious-Plugin-OpenAPI-1.30-1/blib/lib/Mojolicious/Plugin/OpenAPI.pm line 58
	Mojolicious::Plugin::OpenAPI::register(Mojolicious::Plugin::OpenAPI=HASH(0x47e8b660), Mojolicious::Lite=HASH(0x41a8a480), HASH(0x41a25a08)) called at /home/cpansand/.cpan/build/2018060612/Mojolicious-7.83-0/blib/lib/Mojolicious/Plugins.pm line 46
	Mojolicious::Plugins::register_plugin(Mojolicious::Plugins=HASH(0x47fc6180), "OpenAPI", Mojolicious::Lite=HASH(0x41a8a480), HASH(0x41a25a08)) called at /home/cpansand/.cpan/build/2018060612/Mojolicious-7.83-0/blib/lib/Mojolicious.pm line 195
	Mojolicious::plugin(Mojolicious::Lite=HASH(0x41a8a480), "OpenAPI", HASH(0x41a25a08)) called at /home/cpansand/.cpan/build/2018060612/Mojolicious-7.83-0/blib/lib/Mojolicious/Lite.pm line 46
	main::plugin("OpenAPI", HASH(0x41a25a08)) called at t/local.t line 45
t/local.t .............. 
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 
```

Statistical analysis presents two candidates:
```
****************************************************************
Regression 'mod:Mojolicious::Plugin::OpenAPI'
****************************************************************
Name           	       Theta	      StdErr	 T-stat
[0='const']    	      1.0000	      0.0000	98723729929676400.00
[1='eq_1.26']  	      0.0000	      0.0000	   1.54
[2='eq_1.27']  	      0.0000	      0.0000	   0.61
[3='eq_1.28']  	      0.0000	      0.0000	   1.00
[4='eq_1.29']  	     -1.0000	      0.0000	-12852734887509540.00
[5='eq_1.30']  	     -1.0000	      0.0000	-42243904980142000.00

R^2= 1.000, N= 82, K= 6
****************************************************************

****************************************************************
Regression 'mod:JSON::Validator'
****************************************************************
Name           	       Theta	      StdErr	 T-stat
[0='const']    	      1.0000	      0.0000	51826143890780664.00
[1='eq_2.02']  	      0.0000	      0.0000	   4.07
[2='eq_2.03']  	      0.0000	      0.0000	   5.04
[3='eq_2.05']  	      0.0000	      0.0000	   3.22
[4='eq_2.06']  	      0.0000	      0.0000	   2.49
[5='eq_2.07']  	      0.0000	      0.0000	   5.94
[6='eq_2.08']  	     -1.0000	      0.0000	-38783134823389712.00

R^2= 1.000, N= 82, K= 7
****************************************************************
```

**Comment by eserte** [comment]
New statistical data: it seems that it's caused only by JSON::Validator 2.08.

**Comment by mohawk2** [comment]
Thanks for the report! @jhthorsen any ideas?

**Comment by jhthorsen** [comment]
I'm not sure about the specifics since I haven't read the test, but I'm thinking it's because of this fix: https://github.com/jhthorsen/json-validator/issues/103

**Comment by mohawk2** [comment]
@jhthorsen Turns out JV was quite right. Previously `t/local.t` had a `query` parameter whose type was a `$ref`, which is forbidden in OAI 2: https://swagger.io/docs/specification/2-0/describing-parameters/#query-parameters . Fixed in 0.12. Thanks again, @eserte !
